### PR TITLE
Fix a uptr<char> to uptr<char[]> in all functions

### PR DESCRIPTION
**Pull request by umar456**

### CODE REVIEW

File: src/backend/cuda/jit.cpp [hunk 1]
```diff
@@ -13,12 +13,12 @@
 #include <JIT/Node.hpp>
 
 #include <ptx_headers/arith.hpp>
-#include <ptx_headers/logic.hpp>
+#include <ptx_headers/cast.hpp>
 #include <ptx_headers/exp.hpp>
+#include <ptx_headers/hyper.hpp>
+#include <ptx_headers/logic.hpp>
```
Comment (pavanky) [inline src/backend/cuda/jit.cpp]: alphabetical order :+1:

File: src/backend/cuda/jit.cpp [hunk 2]
```diff
@@ -446,10 +449,10 @@ static char *irToPtx(string IR, size_t *ptx_size)
 
     NVVM_CHECK(nvvmGetCompiledResultSize(prog, ptx_size), "Can not get ptx size");
 
-    char *ptx = new char[*ptx_size];
-    NVVM_CHECK(nvvmGetCompiledResult(prog, ptx), "Can not get ptx from NVVM IR");
+    unique_ptr<char[]> ptx{new char[*ptx_size]};
+    NVVM_CHECK(nvvmGetCompiledResult(prog, ptx.get()), "Can not get ptx from NVVM IR");
     NVVM_CHECK(nvvmDestroyProgram(&prog), "Failed to destroy program");
-    return ptx;
+    return ptx.release();
```
Comment (pavanky) [inline src/backend/cuda/jit.cpp]: Is this necessary? Can't we return a `unique_ptr` instead of returning a `char *`?

Comment (umar456) [inline src/backend/cuda/jit.cpp]: Good point. Done.

**Comment by umar456** [comment]
Addressed all feedback.

**Pull request merged**